From f91341fe3f993e73ccfae8b567a55b7b520f8431 Mon Sep 17 00:00:00 2001
From: Jaeyoon Jung <jaeyoon.jung@lge.com>
Date: Thu, 8 Feb 2018 14:24:20 +0900
Subject: [PATCH] Warn circular dependency when loading types

Detect a circular dependency between two QML types and print a warning
message if found.

Change-Id: I826b28775ba3212ccccaeb56d50b64fec6f38ea9
Upstream-Status: Pending
---
 src/qml/qml/qqmltypeloader.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/qml/qml/qqmltypeloader.cpp b/src/qml/qml/qqmltypeloader.cpp
index e9883c0..6d20955 100644
--- a/src/qml/qml/qqmltypeloader.cpp
+++ b/src/qml/qml/qqmltypeloader.cpp
@@ -448,6 +448,10 @@ void QQmlDataBlob::addDependency(QQmlDataBlob *blob)
 
     m_waitingFor.append(blob);
     blob->m_waitingOnMe.append(this);
+
+    // Check circular dependency
+    if (m_waitingOnMe.indexOf(blob) >= 0)
+        qWarning() << "Detected a circular dependency between" << this->url().toString() << "and" << blob->url().toString();
 }
 
 /*!
